freqtrade/freqtradebot.py

```diff
@@ -1339,13 +1340,13 @@ def _safe_exit_amount(self, pair: str, amount: float) -> float:
         :return: amount to sell
         :raise: DependencyException: if available balance is not within 2% of the available amount.
         """
-        # TODO-lev Maybe update?
         # Update wallets to ensure amounts tied up in a stoploss is now free!
         self.wallets.update()
         trade_base_currency = self.exchange.get_pair_base_currency(pair)
         wallet_amount = self.wallets.get_free(trade_base_currency)
         logger.debug(f"{pair} - Wallet: {wallet_amount} - Trade-amount: {amount}")
-        if wallet_amount >= amount:
+        if wallet_amount >= amount or self.trading_mode is TradingMode.FUTURES:
+            # A safe exit amount isn't needed for futures, you can just exit/close the position
```

Comment:
I think this introduces a bug, while hiding bigger problems.

First of all, the balance (`/balance` via telegram) will be reported wrong (this is failing ONLY because the balance is not correct - otherwise it'd be working as is) - and will be missing the position that is open, most likely understating overall balance (so if you've got 100% of your capital in positions, /balance will report 0 balance).

It'll also make freqtrade fail/crash/error if the position was closed on-exchange by the user (this is a scenario that's quite likely - and we need to account for).

Reply:
What should change?

Reply:
in my understanding, you introduced this line to avoid having exiting fail because "balance is too low" (or whatever the error is called).
That might even be fine (i'm not sure it is - but i guess we can leave it as workaround for now).

To me, the proper fix is however to actually have balance show up "somehow" in our wallet (it's a future we own - but we still own it - and currently it'll look like you're broke if you're in a position).

i'm not sure yet how to approach this in detail - but `fetch_balance()` will not return the amount of the positions (at least not for all exchanges).
We might therefore need to have an alternative way to get the proper balance (i'd however do that completely separate, not in this PR).

an example of the problem:
i have a wallet of 1000$
i enter a long position with 2x leverage and go "all in" - so now i have a position of 2000$ in ETH.
if i now run `/balance` - i'll have "0.0" as my balance - since the balance is not available - and positions don't show up. Instead, i should have 2000 ETH (maybe with a hint that it's a 2x position) ... 


Having this would automatically/implicitly fix this problem - and is needed for correct "/balance" output, too.

Reply:
Where's the part that needs to change? We could just call `fetchPositions` and sum (contracts*contractSize*price), we would have to call `fetchTickers` also if we want the real price instead of the markPrice

Reply:
i don't know (as said above, i'm not sure how to approach this yet).
it'll be either wallet updating - or in the exchange (some addition to the get_balances method) ... which is really just syntactic suggar (where you put it).

i think it'll depend how we want/need it - as i'm not sure we can/should add position amounts directly to "ETH free" - as that'd be technically wrong (we do have an ETH position - but we can't use that ETH for anything else ...).
 
Maybe for this PR - add it as a TODO to the above line (in my understanding that line should be done differently once balance reporting is corrected) - and add it as a todo task in the dashboard.

Reply:
@xmatthias Your idea could work, I went after the endpoint what we need, if we can use `fetchPositions` , that should give back the `'conracts` which is the number of asset what we bought. Proly if we put that number there, that should work. 
![image](https://user-images.githubusercontent.com/38407205/152341838-bbe6d8c2-02d9-4769-aebb-0e588c1e6af0.png) So in your example we will get information about how many ETH we bought for that 2000 USDT. in this way we don't need any problematic calculation.